packaging/docker/sidecar.py

```diff
@@ -226,7 +234,7 @@ def __init__(self):
             self.substitutions["FDB_MACHINE_ID"] = os.getenv("HOSTNAME", "")
 
         if self.substitutions["FDB_ZONE_ID"] == "":
-            self.substitutions["FDB_ZONE_ID"] = self.substitutions["FDB_MACHINE_ID"]
+            self.substitutions["FDB_ZONE_ID"] = self.read_node_zone_topology_label(self, self.substitutions["FDB_MACHINE_ID"])
```

Comment:
Based on the discussion in the PR for the operator that should probably be another key instead of zone e.g. data_hall or datacenter to mimic FDBs three_data_hall (or three_data_center) replication mode, that requires the data hall locality and the zone locality (where zone is normally a machine or rack, depending on the topology).

If we could generalise that and have an input like `--read-node-label TARGET_SUBSTITUTION=failure-domain.beta.kubernetes.io/zone` that, we could use this setup for different variable substations and not only for a predefined set. What is your thought on that?

Reply:
Nice idea. Sounds good to me. Flexibility all the way :D